src/gen_ecmult_gen_static_prec_table.c

```diff
@@ -51,38 +44,42 @@ int main(int argc, char **argv) {
 
     fprintf(fp, "#define SC SECP256K1_GE_STORAGE_CONST\n");
 
-    fprintf(fp, "#if ECMULT_GEN_PREC_BITS != %d\n", bits);
-    fprintf(fp, "   #error configuration mismatch, invalid ECMULT_GEN_PREC_BITS. Try deleting ecmult_static_context.h before the build.\n");
-    fprintf(fp, "#endif\n");
-
     fprintf(fp, "static\n");
     fprintf(fp, "#ifndef EXHAUSTIVE_TEST_ORDER\n");
     fprintf(fp, "const\n");
     fprintf(fp, "#endif\n");
     fprintf(fp, "secp256k1_ge_storage secp256k1_ecmult_gen_prec_table[ECMULT_GEN_PREC_N(ECMULT_GEN_PREC_B)][ECMULT_GEN_PREC_G(ECMULT_GEN_PREC_B)] = {\n");
 
-    table = checked_malloc(&default_error_callback, ECMULT_GEN_PREC_TABLE_SIZE);
-    secp256k1_ecmult_gen_create_prec_table(table, &secp256k1_ge_const_g, bits);
-
-    for(outer = 0; outer != n; outer++) {
-        fprintf(fp,"{\n");
-        for(inner = 0; inner != g; inner++) {
-            fprintf(fp,"    SC(%uu, %uu, %uu, %uu, %uu, %uu, %uu, %uu, %uu, %uu, %uu, %uu, %uu, %uu, %uu, %uu)", SECP256K1_GE_STORAGE_CONST_GET(table[outer * g + inner]));
-            if (inner != g - 1) {
-                fprintf(fp,",\n");
+    for (bits = 2; bits <= 8; bits += 2) {
```

Comment:
What's the motivation for `bits += 2` instead of `bits *= 2` here, given that `ecmult_gen.h` has the following?

```
#if ECMULT_GEN_PREC_BITS != 2 && ECMULT_GEN_PREC_BITS != 4 && ECMULT_GEN_PREC_BITS != 8
#  error "Set ECMULT_GEN_PREC_BITS to 2, 4 or 8."
#endif
```

Does either this loop or the check in `ecmult_gen.h` need to be updated?

Reply:
Just noticed this too; it adds a table for ECMULT_GEN_PREC_BITS=6, which isn't allowed.

Reply:
Lol, will fix this.  That's great, so the file is even shorter!

Reply:
Went down from 1.8 MB to 1.4 MB